### PR TITLE
Add max_hold_days early exit parameter

### PR DESCRIPTION
**Pull request by michaelchu**
## Summary
- Adds `max_hold_days` parameter to cap how long a position stays open (calendar days), regardless of P&L
- Piggybacks on the existing early-exit infrastructure (stop-loss/take-profit) as a third trigger condition
- Priority when multiple triggers fire on the same date: stop_loss > take_profit > max_hold
- Works for single-leg, multi-leg, and calendar/diagonal strategies

## Test plan
- [x] Single-leg: triggers at correct date, no-trigger when hold period < max_hold_days, first eligible date
- [x] Multi-leg: spread exits at max_hold_days
- [x] Combined: stop_loss/take_profit fires before max_hold, max_hold fires before stop_loss
- [x] Validation: rejects 0, negative, and float values
- [x] Full test suite passes (1543 tests)

### CODE REVIEW

File: optopsy/exits.py [hunk 1]
```diff
@@ -207,6 +217,21 @@ def _apply_multi_leg_exits(
...
+        "quote_date_entry_leg1",
+    ]:
+        if col in result.columns:
+            entry_date_col = col
+            break
+    if entry_date_col is not None:
+        trade_dates = result[["_trade_id", entry_date_col]].rename(
+            columns={entry_date_col: "_entry_date"}
+        )
+        combined = combined.merge(trade_dates, on="_trade_id")
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/exits.py:235]: Valid catch. Added `"quote_date"` as a fallback in the entry date column lookup for calendar strategies. Note: the broader calendar early-exit pipeline (`_apply_multi_leg_exits`) has a pre-existing column mismatch issue (`expiration` vs `expiration_leg1`, etc.) that affects stop_loss/take_profit too — that's a separate fix.

File: tests/test_early_exits.py [hunk 2]
```diff
@@ -352,3 +352,142 @@ def test_take_profit_must_be_float(self, multi_date_data):
...
+            leg1_delta={"target": 0.50, "min": 0.40, "max": 0.60},
+            leg2_delta={"target": 0.30, "min": 0.10, "max": 0.40},
+            max_hold_days=10,
+            raw=True,
+        )
+        assert not result.empty
+        assert "exit_type" in result.columns
+        triggered = result[result["exit_type"] == "max_hold"]
+        assert not triggered.empty
+
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_early_exits.py:418]: Agreed on the gap. The entry date fallback is now fixed (added `quote_date` to the lookup chain). However, a full calendar strategy early-exit test is blocked by a pre-existing column mismatch in `_apply_multi_leg_exits` (calendar uses `expiration_leg1`/`expiration_leg2` instead of `expiration`). Will address in a follow-up.

File: optopsy/exits.py [hunk 3]
```diff
@@ -334,13 +374,14 @@ def _find_first_threshold_crossing(
...
-    # Determine exit type — if both thresholds could trigger, check which one
-    # When both trigger on the same date, stop_loss takes priority (conservative)
+    # Determine exit type — priority: stop_loss > take_profit > max_hold
     def _classify(row):
         pct = row["_unrealized_pct"]
         if stop_loss is not None and pct <= stop_loss:
             return "stop_loss"
-        return "take_profit"
+        if take_profit is not None and pct >= take_profit:
+            return "take_profit"
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/exits.py:385]: Good call — added the guard. `_classify` now raises `ValueError` if it falls through without `max_hold_days` being set, preventing silent misclassification from float precision edge cases.

File: tests/test_early_exits.py [hunk 4]
```diff
@@ -352,3 +352,142 @@ def test_take_profit_must_be_float(self, multi_date_data):
...
+        result = op.long_puts(
+            multi_date_data,
+            leg1_delta={"target": 0.30, "min": 0.15, "max": 0.70},
+            take_profit=0.25,
+            max_hold_days=15,
+            raw=True,
+        )
+        assert not result.empty
+        assert result["exit_type"].iloc[0] == "take_profit"
+
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_early_exits.py:463]: Added `test_stop_loss_and_max_hold_same_date`: stop_loss=-0.25 and max_hold_days=5 both trigger at day 5, asserts stop_loss wins.

[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_early_exits.py:463]: Added `test_take_profit_and_max_hold_same_date`: take_profit=0.25 and max_hold_days=5 for long puts both trigger at day 5, asserts take_profit wins.

[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_early_exits.py:463]: Added `test_all_three_triggers_priority`: sets stop_loss=-0.25, take_profit=5.0, max_hold_days=5. Stop_loss and max_hold both trigger at day 5, asserts stop_loss wins (full priority chain verified).

File: tests/test_early_exits.py [hunk 5]
```diff
@@ -352,3 +352,142 @@ def test_take_profit_must_be_float(self, multi_date_data):
...
+            )
+
+    def test_max_hold_days_rejects_negative(self, multi_date_data):
+        """max_hold_days must be > 0."""
+        with pytest.raises(Exception):
+            op.long_calls(
+                multi_date_data,
+                leg1_delta={"target": 0.30, "min": 0.15, "max": 0.40},
+                max_hold_days=-5,
+            )
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_early_exits.py]: Agreed — combined into a single parametrized test `test_max_hold_days_rejects_non_positive` with `@pytest.mark.parametrize("value", [0, -5])`.
